### PR TITLE
Setup API Hook, add prettier, add basic JS type-checking with TS

### LINKED ISSUE #87: Add Standard Code Style
Having every file have code styles based on who last edited it is suboptimal. We should add a configuration that automatically applies formatting on commit.

### LINKED ISSUE #88: Add hooks for calling API
Without using a reusable hook, every request needs highly verbose code written inside a useEffect. Propose using axios + swr to make this convenient.

### LINKED ISSUE #89: Add rudimentary typescript support
We can use TS typechecking in JS through doc comments. This proposes adding a tsconfig that will allow for basic type checking when desired (such as on widely used hooks and functions).

### PR DESCRIPTION
**Pull request by frasermcc9**
# Description

- Adds a `useApi` and `useApiMutation` hook.
  - Uses Axios and SWR internally.
- Setup prettier and husky to apply prettier formatting when staging.
- Add a tsconfig that can perform basic typechecking on js files.

## Related Issue

Resolves #87 
Resolves #88 
Resolves #89 

## Type of change

- [X] New feature (enhancement)
- [X] Refactor

## How Has This Been Tested?

- [X] Manual testing

# Checklist:

- [ ] Does a similar (open or closed) pull request not already exist?
- [ ] Is the pull request _head_ repository a fork repository?
- [ ] Is the pull request _compare_ branch a development branch?
- [ ] Is the code documented, particularly in hard-to-understand areas?
- [ ] Does the code build without new warnings?
- [ ] Has testing been performed that proves changes are effective and work?
- [ ] Has a self- and/or peer-review of the code been performed?
- [ ] Have dependent changes been merged and published in downstream modules?
- [ ] Does all new and existing automated testing pass?
- [ ] Is the person responsible for the repository assigned to the pull request?
- [ ] Is the pull request linked to a project?
- [ ] Is the pull request linked to a milestone?

For more information, refer to the Contributing Guidelines and Code of Conduct links at the bottom of this page.

### CODE REVIEW

File: src/hooks/useApi.js [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+};
+
+/**
+ *
+ * @param {string} url
+ * @param {Record<string, string | number>} params
+ * @returns {string}
+ */
+export const substitutePathParams = (url, params = {}) => {
+  const regex = /\{.*?\}/g;
```
Comment (GlockTian) [inline src/hooks/useApi.js]: What is this means? Shall we put a comment on it?

**Review by Sparye — DISMISSED** [review]
A lot of the changes are just styling. useApi seems fine.

File: src/hooks/useApi.js (hunk 1 above)
Comment (frasermcc9) [reply src/hooks/useApi.js]: Sorry this code was copied from the last useApi but i ended up deciding to not use it.

File: src/hooks/useApi.js [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+ * @param {string} url
+ * @returns {string}
+ */
+export const getUrl = (url) => {
+  if (!url.startsWith("/")) {
+    url = "/" + url;
+  }
+
+  const base =
+    process.env.NODE_ENV === "development"
```
Comment (GlockTian) [inline src/hooks/useApi.js:59]: I remembered there is a similar line in src/api/index.js,
did that config anything? if not should we just remove it?

**Review by GlockTian — APPROVED** [review]

File: src/hooks/useApi.js [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+ */
+export const useMutation = (url, options) => {
+  const fn = useCallback(
+    /**
+     * @param {import("axios").AxiosRequestConfig} overrideOptions
+     */
+    (overrideOptions) => {
+      return fetcher({
+        ...{ ...options, ...overrideOptions },
+      })(url);
```
Comment (wshe874) [inline src/hooks/useApi.js]: Just wondering why we don't include url as a part of the parameter to the fetcher function?

Comment (frasermcc9) [reply src/hooks/useApi.js]: SWR passes the URL into the function returned by fetcher

**Pull request merged**